src/battle_util.c

```diff
@@ -4611,12 +4605,24 @@ u32 AbilityBattleEffects(u32 caseID, u32 battler, u32 ability, u32 special, u32
             }
             break;
         case ABILITY_SUPREME_OVERLORD:
-            if (!gSpecialStatuses[battler].switchInAbilityDone && CountUsablePartyMons(battler) < PARTY_SIZE)
+            if (!gSpecialStatuses[battler].switchInAbilityDone)
             {
+                struct Pokemon* party = GetBattlerParty(battler);
                 gSpecialStatuses[battler].switchInAbilityDone = TRUE;
-                gBattleStruct->supremeOverlordModifier[battler] = GetSupremeOverlordModifier(battler);
-                BattleScriptPushCursorAndCallback(BattleScript_SupremeOverlordActivates);
-                effect++;
+                j = 0;
+                for (i = 0; i < PARTY_SIZE; i++)
```

Comment:
Why was this added? The pre-existing faint counter should be used.

Reply:
It had been so long since I did research that I forgot that I assumed stuff based on Bulbapedia's description. It should've used the same counter as Last Respects.
https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9421520
Will fix it.